packages/typescript-estree/tests/lib/parse.moduleResolver.default-program-error.test.ts

```diff
@@ -0,0 +1,52 @@
+import { join, resolve } from 'path';
+
+import * as parser from '../../src';
+import type { TSESTreeOptions } from '../../src/parser-options';
+
+const FIXTURES_DIR = join(__dirname, '../fixtures/simpleProject');
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('parseAndGenerateServices', () => {
+  describe('moduleResolver', () => {
+    beforeEach(() => {
+      parser.clearCaches();
+    });
+
+    const PROJECT_DIR = resolve(FIXTURES_DIR, '../moduleResolver');
+    const code = `
+      import { something } from '__PLACEHOLDER__';
+
+      something();
+    `;
+    const config: TSESTreeOptions = {
+      comment: true,
+      tokens: true,
+      range: true,
+      loc: true,
+      project: './tsconfig.json',
+      tsconfigRootDir: PROJECT_DIR,
+      filePath: resolve(PROJECT_DIR, 'file.ts'),
+    };
```

Comment:
I'm seeing some duplication still 😄. Each of the new test files has roughly the same setup:
1. `beforeEach(...)`
2. Creating identical `PROJECT_DIR`, `code`, and `config` values
3. Passing `...config` with some values (either inline or from a variable) to `parser.parseAndGenerateServices`

How about moving all that setup to a shared function? I sketched it out locally and it all still seems to work.

```ts
export const createAndPrepareParseConfig = () => {
  beforeEach(...);

  const projectDirectory = ...
  const code = ...
  const config = ...

  return { code, config, projectDirectory };
};
```

Reply:
Apologies for the very late response, thank you for your patience and the suggestion!

Actioned in https://github.com/typescript-eslint/typescript-eslint/pull/6092/commits/77b38fc20c2e8ff23dfc8b69aafda5b72ea75b0e ✅